### DISCUSSION TITLE
Question on fc.float() and fc.double()

### DISCUSSION THREAD

**Discussion by JamieMason** [dubzzz/fast-check#3658] [Q&A] [answered]
I'm probably misunderstanding something or doing something wrong here 😬  Can anyone spot what I'm missing? Thanks.

## Steps to reproduce

Run the following

```js
const fc = require('fast-check');

fc.assert(
  fc.property(fc.oneOf(fc.float(), fc.double()), (value) => {
    return String(value).indexOf('.') !== -1
  })
);
```

## Expected behaviour

`value` should always have a decimal place(?)

## Actual behaviour

`value` is sometimes an integer, for example:

```
    Property failed after 1 tests
    { seed: -1373937679, path: "0:2:0:1:2:1:0:1:2:1:0:1:1", endOnFailure: true }
    Counterexample: [-8388563]
    Shrunk 12 time(s)
```

**Comment by dubzzz** [comment, marked as answer]
What we call `float` in fast-check refers to [single precision floating point](https://en.wikipedia.org/wiki/Single-precision_floating-point_format), while what we call `double` refers to [double precision floating point](https://en.wikipedia.org/wiki/Double-precision_floating-point_format), so both are able to generate integer values as they are part of their scope.

For now, we do not provide any `decimal` allowing users to say: "I want at least N decimals".

**Reply by JamieMason** [reply]
Got it, thanks Nicolas
